pkg/controller/infrastructure/actuator_reconcile.go

```diff
@@ -18,30 +18,47 @@ import (
 	"context"
 
 	apiskubevirt "github.com/gardener/gardener-extension-provider-kubevirt/pkg/apis/kubevirt"
+	"github.com/gardener/gardener-extension-provider-kubevirt/pkg/apis/kubevirt/helper"
 	apiskubevirtv1alpha1 "github.com/gardener/gardener-extension-provider-kubevirt/pkg/apis/kubevirt/v1alpha1"
+	"github.com/gardener/gardener-extension-provider-kubevirt/pkg/internal/infrastructure"
 
 	extensionscontroller "github.com/gardener/gardener/extensions/pkg/controller"
 	extensionsv1alpha1 "github.com/gardener/gardener/pkg/apis/extensions/v1alpha1"
+
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/client-go/tools/clientcmd/api"
 	"k8s.io/client-go/util/retry"
+
+	"github.com/pkg/errors"
 )
 
 func (a *actuator) Reconcile(ctx context.Context, infra *extensionsv1alpha1.Infrastructure, cluster *extensionscontroller.Cluster) error {
-	// TODO: here, controller should ensure the network setup
+	config, err := helper.InfrastructureConfigFromInfrastructure(infra)
+	if err != nil {
+		return err
+	}
+	// retrieve pre-created net-attach-def from reference and set to the infrastructure status
+	// TODO if net-attach-def don't exist, create them using IPAM config
+	nadRefs := config.NetAttachDefRef
+	nads, err := infrastructure.GetNetworkAttachmentDefinition(ctx, a.Client(), &nadRefs)
```

Comment:
This `a.Client()` will return the client for the seed cluster. Do you need to read the network attachment definitions from the seed or the provider cluster? These clusters are one and the same in our current test setup, but in fact they can be different. If you need to work with the provider cluster, you might need to use the `GetCredentials` function you defined elsewhere to get the kubeconfig of the provider cluster.

Reply:
now use provider's kubeconfig and get net-attach-def from there.